services/datalad/datalad_service/handlers/annex_objects.py

```diff
@@ -26,6 +28,7 @@ def on_delete(self, req, resp, dataset, snapshot, annex_key):
             urls = file.get('urls')
 
             gevent.spawn(remove_file_remotes, self.store, urls)
+            FilesResource.on_delete(self, req, resp, dataset, filename)
```

Comment:
This generates a new commit, which is not the intention of removing the annexed content.

You don't need the filename at all to accomplish this, instead just remove the annex object file. It's probably best to just call `git-annex drop --force --key=` to do this.